src/System.Threading.Timer/tests/TimerChangeTests.cs

```diff
@@ -2,115 +2,82 @@
-    }
 
-    [Fact]
-    public void Timer_Change_TooLongTimeSpan_Period_Throws()
-    {
-        using (var t = new Timer(new TimerCallback(EmptyTimerTarget), null, 1, 1))
+        [Fact]
+        public void Timer_Change_AfterDispose_Throws()
         {
-            Assert.Throws<ArgumentOutOfRangeException>(() => t.Change(new TimeSpan(1) /* not relevant */, TimeSpan.FromMilliseconds((long)0xFFFFFFFF)));
+            var t = new Timer(new TimerCallback(EmptyTimerTarget), null, 1, 1);
+            t.Dispose();
+            Assert.Throws<ObjectDisposedException>(() => t.Change(1, 1));
+            Assert.Throws<ObjectDisposedException>(() => t.Change(1L, 1L));
+            Assert.Throws<ObjectDisposedException>(() => t.Change(1u, 1u));
+            Assert.Throws<ObjectDisposedException>(() => t.Change(TimeSpan.FromMilliseconds(1), TimeSpan.FromMilliseconds(1)));
         }
-    }
-
-    [Fact]
-    public void Timer_Change_TimeSpan_AfterDispose_Throws()
-    {
-        var t = new Timer(new TimerCallback(EmptyTimerTarget), null, 1, 1);
-        t.Dispose();
-        Assert.Throws<ObjectDisposedException>(() => t.Change(TimeSpan.FromMilliseconds(1), TimeSpan.FromMilliseconds(1)));
-    }
-
-    [Fact]
-    public void Timer_Change_Int_AfterDispose_Throws()
-    {
-        var t = new Timer(new TimerCallback(EmptyTimerTarget), null, 1, 1);
-        t.Dispose();
-        Assert.Throws<ObjectDisposedException>(() => t.Change(1, 1));
-    }
-
-    [Fact]
-    public void Timer_Change_BeforeDueTime_ChangesWhenTimerWillFire()
-    {
-        AutoResetEvent are = new AutoResetEvent(false);
 
-        using (var t = new Timer(new TimerCallback((object s) =>
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(2)]
+        [InlineData(3)]
+        public void Timer_Change_BeforeDueTime_ChangesWhenTimerWillFire(int mode)
         {
-            are.Set();
-        }), null, TimeSpan.FromSeconds(500), TimeSpan.FromMilliseconds(50)))
-        {
-            Assert.False(are.WaitOne(TimeSpan.FromMilliseconds(100)), "The reset event should not have been set yet");
-            t.Change(TimeSpan.FromMilliseconds(100), TimeSpan.FromMilliseconds(-1));
-            Assert.True(are.WaitOne(TimeSpan.FromMilliseconds(TimerFiringTests.MaxPositiveTimeoutInMs)), "Should have received a timer event after this new duration");
-        }
-    }
-
-    [Fact]
-    public void Timer_Change_Int64_Negative()
-    {
-        Assert.Throws<ArgumentOutOfRangeException>(() => new Timer(EmptyTimerTarget).Change((long)-2, (long)-1));
-        Assert.Throws<ArgumentOutOfRangeException>(() => new Timer(EmptyTimerTarget).Change((long)-1, (long)-2));
-        Assert.Throws<ArgumentOutOfRangeException>(() => new Timer(EmptyTimerTarget).Change((long)0xffffffff, (long)-1));
-        Assert.Throws<ArgumentOutOfRangeException>(() => new Timer(EmptyTimerTarget).Change((long)-1, (long)0xffffffff));
-    }
+            var are = new AutoResetEvent(false);
```

Comment:
Dispose AutoResetEvent ?

Reply:
The problem with that in these async tests is that if someone goes wrong and the timer callback end up executing after the are is disposed, we'll get disposed exceptions. It's probably not an issue here specifically, but that's why the rest of the tests leave it for finalization, and got consistency I've done the same.